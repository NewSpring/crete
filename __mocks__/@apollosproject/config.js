const ApolloServer = require.requireActual('@apollosproject/config').default;

ApolloServer.loadJs({
  ROCK: {
    API_URL: 'https://apollosrock.newspring.cc/api',
    API_TOKEN: 'some-rock-token',
    IMAGE_URL: 'https://apollosrock.newspring.cc/GetImage.ashx',
    SHARE_URL: 'https://newspring.cc',
    TIMEZONE: 'America/New_York',
  },
  ROCK_CONSTANTS: {
    IMAGE: 10,
    AUDIO_FILE: 77,
    VIDEO_FILE: 79,
  },
  ROCK_MAPPINGS: {
    FEED_CONTENT_CHANNEL_IDS: [1, 2, 3, 4, 6, 8],
    SERIES_CONTENT_CHANNEL_TYPE_IDS: [6, 7],
    CONTENT_ITEM: {
      ContentSeriesContentItem: {
        EntityType: 'ContentChannelItem',
        ContentChannelId: [18, 20],
      },
      DevotionalContentItem: {
        EntityType: 'ContentChannelItem',
        ContentChannelId: [17],
      },
      MediaContentItem: {
        EntityType: 'ContentChannelItem',
      },
      UniversalContentItem: {
        EntityType: 'ContentChannelItem',
      },
      ContentItem: {
        EntityType: 'ContentChannelItem',
      },
      WeekendContentItem: {
        EntityType: 'ContentChannelItem',
      },
    },
  },
});

export default ApolloServer;
