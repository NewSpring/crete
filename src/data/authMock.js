export default class Auth {
  initialize = () => {};

  getCurrentPerson = () => ({
    id: 51,
    primaryAliasId: 123,
    firstName: 'Isaac',
    lastName: 'Hardy',
    nickName: 'Isaac',
    email: 'isaac.hardy@newspring.cc',
    photo: {
      url:
        'https://apollosrock.newspring.cc:443/GetImage.ashx?guid=60fd5f35-3167-4c26-9a30-d44937287b87',
    },
  });
}
