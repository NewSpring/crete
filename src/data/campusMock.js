export default class Campus {
  getFromId = () => ({
    id: 16,
    name: 'Anderson',
  });
}
