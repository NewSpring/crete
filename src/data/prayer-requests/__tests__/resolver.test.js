import { graphql } from 'graphql';
import { fetch } from 'apollo-server-env';
import { createTestHelpers } from '@apollosproject/server-core/lib/testUtils';

import { peopleSchema, campusSchema } from '@apollosproject/data-schema';
import * as PrayerRequest from '../index';

import prayerRequestSchema from '../schema';
import authMock from '../../authMock';
import campusMock from '../../campusMock';

const { getSchema, getContext } = createTestHelpers({
  PrayerRequest,
  Auth: { dataSource: authMock },
  Person: { dataSource: authMock },
  Campus: { dataSource: campusMock },
});

const currentPersonResMock = jest.fn(() =>
  Promise.resolve({
    id: 1,
    firstName: 'Isaac',
    lastName: 'Hardy',
  })
);

const onePrayerResMock = jest.fn(() =>
  Promise.resolve({
    id: 'PrayerRequest:b36e55d803443431e96bb4b5068147ec',
    firstName: 'Isaac',
    lastName: 'Hardy',
    text: 'Pray this works.',
    requestedByPersonAliasId: 447217,
    campusId: 16,
    categoryId: 2,
    flagCount: 0,
    prayerCount: 4,
    attributeValues: {
      isAnonymous: {
        value: 'True',
      },
    },
  })
);

const twoPrayerResMock = jest.fn(() =>
  Promise.resolve([
    {
      id: 'PrayerRequest:b36e55d803443431e96bb4b5068147ec',
      firstName: 'Isaac',
      lastName: 'Hardy',
      text: 'Pray this works.',
      requestedByPersonAliasId: 447217,
      enteredDateTime: '2019-07-02T13:08:02.035',
      campusId: 16,
      categoryId: 2,
      flagCount: 0,
      prayerCount: 4,
      attributeValues: {
        isAnonymous: {
          value: 'True',
        },
      },
    },
    {
      id: 'PrayerRequest:57c465ee3cd69524d729569b338607de',
      firstName: 'Rich',
      lastName: 'Dubee',
      text: 'Help me',
      requestedByPersonAliasId: 447217,
      enteredDateTime: '2019-07-03T13:08:02.035',
      campusId: 16,
      categoryId: 2,
      flagCount: 0,
      prayerCount: 4,
      attributeValues: {
        isAnonymous: {
          value: 'True',
        },
      },
    },
  ])
);
describe('PrayerRequest resolver', () => {
  let schema;
  let context;
  let rootValue;
  beforeEach(() => {
    fetch.resetMocks();
    fetch.mockRockDataSourceAPI();
    schema = getSchema([prayerRequestSchema, peopleSchema, campusSchema]);
    context = getContext();
    context.dataSources.Person.getFromAliasId = currentPersonResMock;
    context.dataSources.PrayerRequest.createInteraction = jest.fn(() => null);
    rootValue = {};
  });

  it('gets all public prayer requests', async () => {
    const query = `
      query {
        prayers {
          id
          firstName
          lastName
          text
          requestedByPersonAliasId
          campus {
            id
            name
          }
          categoryId
          flagCount
          prayerCount
          isAnonymous
          person {
            id
            firstName
            lastName
          }
        }
      }
    `;

    context.dataSources.PrayerRequest.getAll = twoPrayerResMock;
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('gets all public prayer requests by campus', async () => {
    const query = `
      query {
        campusPrayers {
          id
          firstName
          text
          campus {
            id
            name
          }
        }
      }
    `;
    context.dataSources.PrayerRequest.getAllByCampus = twoPrayerResMock;
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('gets all prayers from current person', async () => {
    const query = `
      query {
        userPrayers {
          id
          firstName
          text
        }
      }
    `;

    context.dataSources.PrayerRequest.getFromCurrentPerson = twoPrayerResMock;
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('gets all prayers from groups', async () => {
    const query = `
      query {
        groupPrayers {
          id
          firstName
          text
        }
      }
    `;

    context.dataSources.PrayerRequest.getFromGroups = twoPrayerResMock;
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('creates a new prayer', async () => {
    const query = `
      mutation {
        addPrayer(
          firstName: "Test"
          lastName: "Bro"
          text: "Jesus Rocks"
          campusId: "Campus:4f68015ba18662a7409d1219a4ce013e"
          categoryId: 1
          isAnonymous: true
        ) {
          id
          firstName
          text
        }
      }
    `;
    context.dataSources.PrayerRequest.add = onePrayerResMock;
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('increments prayed for a request', async () => {
    const query = `
      mutation {
        incrementPrayerCount(
          nodeId: "PrayerRequest:b36e55d803443431e96bb4b5068147ec"
        ) {
          id
          firstName
          text
        }
      }
    `;
    context.dataSources.PrayerRequest.incrementPrayed = onePrayerResMock;
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });

  it('flags a prayer request', async () => {
    const query = `
      mutation {
        flagPrayer(nodeId: "PrayerRequest:b36e55d803443431e96bb4b5068147ec") {
          id
          firstName
          text
        }
      }
    `;
    context.dataSources.PrayerRequest.flag = onePrayerResMock;
    const result = await graphql(schema, query, rootValue, context);
    expect(result).toMatchSnapshot();
  });
});
