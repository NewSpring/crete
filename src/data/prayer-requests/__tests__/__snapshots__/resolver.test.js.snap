// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrayerRequest resolver creates a new prayer 1`] = `
Object {
  "data": Object {
    "addPrayer": Object {
      "firstName": "Isaac",
      "id": "PrayerRequest:4ca828e6155e88cd21a8b67865472a9301e9f68a0e3823859d5a383958bc6c5b843f7453855c22c6f07f5955b5ca0bd1",
      "text": "Pray this works.",
    },
  },
}
`;

exports[`PrayerRequest resolver flags a prayer request 1`] = `
Object {
  "data": Object {
    "flagPrayer": Object {
      "firstName": "Isaac",
      "id": "PrayerRequest:4ca828e6155e88cd21a8b67865472a9301e9f68a0e3823859d5a383958bc6c5b843f7453855c22c6f07f5955b5ca0bd1",
      "text": "Pray this works.",
    },
  },
}
`;

exports[`PrayerRequest resolver gets all prayers from current person 1`] = `
Object {
  "data": Object {
    "userPrayers": Array [
      Object {
        "firstName": "Isaac",
        "id": "PrayerRequest:4ca828e6155e88cd21a8b67865472a9301e9f68a0e3823859d5a383958bc6c5b843f7453855c22c6f07f5955b5ca0bd1",
        "text": "Pray this works.",
      },
      Object {
        "firstName": "Rich",
        "id": "PrayerRequest:04fbd6bf8d5ee1a6493f42615e65ea05bb9ce2844192b00af611268daee3297d6e00b42b4ceebbb632956536f1cb00f3",
        "text": "Help me",
      },
    ],
  },
}
`;

exports[`PrayerRequest resolver gets all prayers from groups 1`] = `
Object {
  "data": Object {
    "groupPrayers": Array [
      Object {
        "firstName": "Isaac",
        "id": "PrayerRequest:4ca828e6155e88cd21a8b67865472a9301e9f68a0e3823859d5a383958bc6c5b843f7453855c22c6f07f5955b5ca0bd1",
        "text": "Pray this works.",
      },
      Object {
        "firstName": "Rich",
        "id": "PrayerRequest:04fbd6bf8d5ee1a6493f42615e65ea05bb9ce2844192b00af611268daee3297d6e00b42b4ceebbb632956536f1cb00f3",
        "text": "Help me",
      },
    ],
  },
}
`;

exports[`PrayerRequest resolver gets all public prayer requests 1`] = `
Object {
  "data": Object {
    "prayers": Array [
      Object {
        "campus": Object {
          "id": "16",
          "name": "Anderson",
        },
        "categoryId": 2,
        "firstName": "Isaac",
        "flagCount": 0,
        "id": "PrayerRequest:4ca828e6155e88cd21a8b67865472a9301e9f68a0e3823859d5a383958bc6c5b843f7453855c22c6f07f5955b5ca0bd1",
        "isAnonymous": true,
        "lastName": "Hardy",
        "person": Object {
          "firstName": "Isaac",
          "id": "1",
          "lastName": "Hardy",
        },
        "prayerCount": 4,
        "requestedByPersonAliasId": 447217,
        "text": "Pray this works.",
      },
      Object {
        "campus": Object {
          "id": "16",
          "name": "Anderson",
        },
        "categoryId": 2,
        "firstName": "Rich",
        "flagCount": 0,
        "id": "PrayerRequest:04fbd6bf8d5ee1a6493f42615e65ea05bb9ce2844192b00af611268daee3297d6e00b42b4ceebbb632956536f1cb00f3",
        "isAnonymous": true,
        "lastName": "Dubee",
        "person": Object {
          "firstName": "Isaac",
          "id": "1",
          "lastName": "Hardy",
        },
        "prayerCount": 4,
        "requestedByPersonAliasId": 447217,
        "text": "Help me",
      },
    ],
  },
}
`;

exports[`PrayerRequest resolver gets all public prayer requests by campus 1`] = `
Object {
  "data": Object {
    "campusPrayers": Array [
      Object {
        "campus": Object {
          "id": "16",
          "name": "Anderson",
        },
        "firstName": "Isaac",
        "id": "PrayerRequest:4ca828e6155e88cd21a8b67865472a9301e9f68a0e3823859d5a383958bc6c5b843f7453855c22c6f07f5955b5ca0bd1",
        "text": "Pray this works.",
      },
      Object {
        "campus": Object {
          "id": "16",
          "name": "Anderson",
        },
        "firstName": "Rich",
        "id": "PrayerRequest:04fbd6bf8d5ee1a6493f42615e65ea05bb9ce2844192b00af611268daee3297d6e00b42b4ceebbb632956536f1cb00f3",
        "text": "Help me",
      },
    ],
  },
}
`;

exports[`PrayerRequest resolver increments prayed for a request 1`] = `
Object {
  "data": Object {
    "incrementPrayerCount": Object {
      "firstName": "Isaac",
      "id": "PrayerRequest:4ca828e6155e88cd21a8b67865472a9301e9f68a0e3823859d5a383958bc6c5b843f7453855c22c6f07f5955b5ca0bd1",
      "text": "Pray this works.",
    },
  },
}
`;
